<html><h1>Validation FAILED.</h1><p>Build Time: 26 minutes 804 ms</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/com-liferay-commerce/tree/7.1.x">7.1.x</a><br/>Branch GIT ID: <a href="https://github.com/liferay/com-liferay-commerce/commit/3055a840437789413d5860119381e170cea40640">3055a840437789413d5860119381e170cea40640</a></p><h6>Task Summary:</h6><ul><li>subrepository-compile-jdk8 -  :heavy_check_mark:</li><li>subrepository-semantic-versioning-jdk8 -  :heavy_check_mark:</li><li>subrepository-pmd-jdk8 -  :heavy_check_mark:</li><li>subrepository-source-format-jdk8 -  :x:<pre><code>     [echo] java.lang.Exception: Found 1 formatting issues:
     [echo] 1: Must include both @java.lang.Deprecated annotation and @deprecated Javadoc tag with description.: ./commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderItemLocalServiceImpl.java 152 (Checkstyle:MissingDeprecatedCheck)
     [echo] 
     [echo] 	at com.liferay.source.formatter.SourceFormatter.format(SourceFormatter.java:439)
     [echo] 	at com.liferay.source.formatter.SourceFormatter.main(SourceFormatter.java:267)
     [echo] 
     [echo] FAILURE: Build failed with an exception.
     [echo] 
     [echo] * What went wrong:
     [echo] Execution failed for task ':checkSourceFormatting'.
     [echo] &gt; Process 'command '/opt/java/jdk1.8.0_121-x64/bin/java'' finished with non-zero exit value 1
     [echo] 
     [echo] * Try:
     [echo] Run with --info or --debug option to get more log output. Run with --scan to get full insights.
     [echo] 
     [echo] * Exception is:
     [echo] org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':checkSourceFormatting'.
     [echo] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:110)
     [echo] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:77)
     [echo] 	at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
     [echo] 	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
     [echo] 	at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
     [echo] 	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59)
     [echo] 	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
     [echo] 	at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
     [echo] 	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:91)
     [echo] 	at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)</code></pre></li><li>central-requirements-jdk8/subrepository-unit-jdk8 -  :heavy_check_mark:</li></ul><h5>For full console, click <a href="https://test-1-21.liferay.com/job/test-subrepository-acceptance-pullrequest-validation(7.1.x)/66//consoleText">here</a>.</h5><hr/><div><h6>Test Results:</h6><div><div><p>48 Tests Passed.<br/>0 Tests Failed.</p></div></div><h5>For all test results, click <a href="https://test-1-21.liferay.com/job/test-subrepository-acceptance-pullrequest-validation(7.1.x)/66//testReport">here</a>.</h5></div></html>

--- a/commerce-api/src/main/java/com/liferay/commerce/service/CommerceOrderItemLocalService.java
+++ b/commerce-api/src/main/java/com/liferay/commerce/service/CommerceOrderItemLocalService.java
@@ -98,8 +98,14 @@ public interface CommerceOrderItemLocalService extends BaseLocalService,
 	* @throws PortalException
 	*/
 	@Indexable(type = IndexableType.DELETE)
+	@Deprecated
 	public CommerceOrderItem deleteCommerceOrderItem(
 		CommerceOrderItem commerceOrderItem) throws PortalException;
+
+	@Indexable(type = IndexableType.DELETE)
+	public CommerceOrderItem deleteCommerceOrderItem(
+		CommerceOrderItem commerceOrderItem, CommerceContext commerceContext)
+		throws PortalException;
 
 	/**
 	* Deletes the commerce order item with the primary key from the database. Also notifies the appropriate model listeners.

--- a/commerce-api/src/main/java/com/liferay/commerce/service/CommerceOrderItemLocalServiceUtil.java
+++ b/commerce-api/src/main/java/com/liferay/commerce/service/CommerceOrderItemLocalServiceUtil.java
@@ -83,10 +83,19 @@ public class CommerceOrderItemLocalServiceUtil {
 	* @return the commerce order item that was removed
 	* @throws PortalException
 	*/
+	@Deprecated
 	public static com.liferay.commerce.model.CommerceOrderItem deleteCommerceOrderItem(
 		com.liferay.commerce.model.CommerceOrderItem commerceOrderItem)
 		throws com.liferay.portal.kernel.exception.PortalException {
 		return getService().deleteCommerceOrderItem(commerceOrderItem);
+	}
+
+	public static com.liferay.commerce.model.CommerceOrderItem deleteCommerceOrderItem(
+		com.liferay.commerce.model.CommerceOrderItem commerceOrderItem,
+		com.liferay.commerce.context.CommerceContext commerceContext)
+		throws com.liferay.portal.kernel.exception.PortalException {
+		return getService()
+				   .deleteCommerceOrderItem(commerceOrderItem, commerceContext);
 	}
 
 	/**

--- a/commerce-api/src/main/java/com/liferay/commerce/service/CommerceOrderItemLocalServiceWrapper.java
+++ b/commerce-api/src/main/java/com/liferay/commerce/service/CommerceOrderItemLocalServiceWrapper.java
@@ -77,11 +77,21 @@ public class CommerceOrderItemLocalServiceWrapper
 	* @return the commerce order item that was removed
 	* @throws PortalException
 	*/
+	@Deprecated
 	@Override
 	public com.liferay.commerce.model.CommerceOrderItem deleteCommerceOrderItem(
 		com.liferay.commerce.model.CommerceOrderItem commerceOrderItem)
 		throws com.liferay.portal.kernel.exception.PortalException {
 		return _commerceOrderItemLocalService.deleteCommerceOrderItem(commerceOrderItem);
+	}
+
+	@Override
+	public com.liferay.commerce.model.CommerceOrderItem deleteCommerceOrderItem(
+		com.liferay.commerce.model.CommerceOrderItem commerceOrderItem,
+		com.liferay.commerce.context.CommerceContext commerceContext)
+		throws com.liferay.portal.kernel.exception.PortalException {
+		return _commerceOrderItemLocalService.deleteCommerceOrderItem(commerceOrderItem,
+			commerceContext);
 	}
 
 	/**

--- a/commerce-api/src/main/java/com/liferay/commerce/service/CommerceOrderItemService.java
+++ b/commerce-api/src/main/java/com/liferay/commerce/service/CommerceOrderItemService.java
@@ -66,6 +66,9 @@ public interface CommerceOrderItemService extends BaseService {
 	public void deleteCommerceOrderItem(long commerceOrderItemId)
 		throws PortalException;
 
+	public void deleteCommerceOrderItem(long commerceOrderItemId,
+		CommerceContext commerceContext) throws PortalException;
+
 	@Transactional(propagation = Propagation.SUPPORTS, readOnly = true)
 	public CommerceOrderItem fetchByExternalReferenceCode(long companyId,
 		String externalReferenceCode) throws PortalException;

--- a/commerce-api/src/main/java/com/liferay/commerce/service/CommerceOrderItemServiceUtil.java
+++ b/commerce-api/src/main/java/com/liferay/commerce/service/CommerceOrderItemServiceUtil.java
@@ -58,6 +58,13 @@ public class CommerceOrderItemServiceUtil {
 		getService().deleteCommerceOrderItem(commerceOrderItemId);
 	}
 
+	public static void deleteCommerceOrderItem(long commerceOrderItemId,
+		com.liferay.commerce.context.CommerceContext commerceContext)
+		throws com.liferay.portal.kernel.exception.PortalException {
+		getService()
+			.deleteCommerceOrderItem(commerceOrderItemId, commerceContext);
+	}
+
 	public static com.liferay.commerce.model.CommerceOrderItem fetchByExternalReferenceCode(
 		long companyId, String externalReferenceCode)
 		throws com.liferay.portal.kernel.exception.PortalException {

--- a/commerce-api/src/main/java/com/liferay/commerce/service/CommerceOrderItemServiceWrapper.java
+++ b/commerce-api/src/main/java/com/liferay/commerce/service/CommerceOrderItemServiceWrapper.java
@@ -52,6 +52,14 @@ public class CommerceOrderItemServiceWrapper implements CommerceOrderItemService
 	}
 
 	@Override
+	public void deleteCommerceOrderItem(long commerceOrderItemId,
+		com.liferay.commerce.context.CommerceContext commerceContext)
+		throws com.liferay.portal.kernel.exception.PortalException {
+		_commerceOrderItemService.deleteCommerceOrderItem(commerceOrderItemId,
+			commerceContext);
+	}
+
+	@Override
 	public com.liferay.commerce.model.CommerceOrderItem fetchByExternalReferenceCode(
 		long companyId, String externalReferenceCode)
 		throws com.liferay.portal.kernel.exception.PortalException {

--- a/commerce-cart-content-web/src/main/java/com/liferay/commerce/cart/content/web/internal/portlet/action/EditCommerceOrderItemMVCActionCommand.java
+++ b/commerce-cart-content-web/src/main/java/com/liferay/commerce/cart/content/web/internal/portlet/action/EditCommerceOrderItemMVCActionCommand.java
@@ -68,7 +68,7 @@ public class EditCommerceOrderItemMVCActionCommand
 		try {
 			if (cmd.equals(Constants.DELETE)) {
 				_commerceOrderItemService.deleteCommerceOrderItem(
-					commerceOrderItemId);
+					commerceOrderItemId, commerceContext);
 			}
 			else if (cmd.equals(Constants.UPDATE)) {
 				int quantity = ParamUtil.getInteger(actionRequest, "quantity");

--- a/commerce-frontend-impl/src/main/java/com/liferay/commerce/frontend/internal/cart/CommerceCartResource.java
+++ b/commerce-frontend-impl/src/main/java/com/liferay/commerce/frontend/internal/cart/CommerceCartResource.java
@@ -105,7 +105,7 @@ public class CommerceCartResource {
 			serviceContext.setScopeGroupId(commerceOrder.getGroupId());
 
 			_commerceOrderItemService.deleteCommerceOrderItem(
-				commerceOrderItem.getCommerceOrderItemId());
+				commerceOrderItem.getCommerceOrderItemId(), commerceContext);
 
 			cart = _commerceCartResourceUtil.getCart(
 				commerceOrder.getCommerceOrderId(), themeDisplay.getLocale(),

--- a/commerce-openapi-admin/src/main/java/com/liferay/commerce/openapi/admin/internal/resource/util/v2_0/OrderItemHelper.java
+++ b/commerce-openapi-admin/src/main/java/com/liferay/commerce/openapi/admin/internal/resource/util/v2_0/OrderItemHelper.java
@@ -45,13 +45,14 @@ import org.osgi.service.component.annotations.Reference;
 @Component(immediate = true, service = OrderItemHelper.class)
 public class OrderItemHelper {
 
-	public void deleteOrderItem(String id, Company company)
+	public void deleteOrderItem(
+			String id, Company company, CommerceContext commerceContext)
 		throws PortalException {
 
 		CommerceOrderItem commerceOrderItem = getOrderItemById(id, company);
 
 		_commerceOrderItemService.deleteCommerceOrderItem(
-			commerceOrderItem.getCommerceOrderId());
+			commerceOrderItem.getCommerceOrderId(), commerceContext);
 	}
 
 	public OrderItemDTO getOrderItem(String id, Company company)

--- a/commerce-openapi-admin/src/main/java/com/liferay/commerce/openapi/admin/internal/resource/v2_0/OrderResourceImpl.java
+++ b/commerce-openapi-admin/src/main/java/com/liferay/commerce/openapi/admin/internal/resource/v2_0/OrderResourceImpl.java
@@ -72,7 +72,8 @@ public class OrderResourceImpl implements OrderResource {
 	public Response deleteOrderItem(String id, String orderItemId)
 		throws Exception {
 
-		_orderItemHelper.deleteOrderItem(orderItemId, _company);
+		_orderItemHelper.deleteOrderItem(
+			orderItemId, _company, _commerceContext);
 
 		Response.ResponseBuilder responseBuilder = Response.noContent();
 

--- a/commerce-order-content-web/src/main/java/com/liferay/commerce/order/content/web/internal/portlet/action/EditCommerceOrderItemMVCActionCommand.java
+++ b/commerce-order-content-web/src/main/java/com/liferay/commerce/order/content/web/internal/portlet/action/EditCommerceOrderItemMVCActionCommand.java
@@ -59,6 +59,10 @@ public class EditCommerceOrderItemMVCActionCommand
 	protected void deleteCommerceOrderItems(ActionRequest actionRequest)
 		throws PortalException {
 
+		CommerceContext commerceContext =
+			(CommerceContext)actionRequest.getAttribute(
+				CommerceWebKeys.COMMERCE_CONTEXT);
+
 		long commerceOrderId = ParamUtil.getLong(
 			actionRequest, "commerceOrderId");
 
@@ -70,7 +74,7 @@ public class EditCommerceOrderItemMVCActionCommand
 
 		for (CommerceOrderItem commerceOrderItem : commerceOrderItems) {
 			_commerceOrderItemService.deleteCommerceOrderItem(
-				commerceOrderItem.getCommerceOrderItemId());
+				commerceOrderItem.getCommerceOrderItemId(), commerceContext);
 		}
 	}
 
@@ -91,7 +95,7 @@ public class EditCommerceOrderItemMVCActionCommand
 		try {
 			if (cmd.equals(Constants.DELETE)) {
 				_commerceOrderItemService.deleteCommerceOrderItem(
-					commerceOrderItemId);
+					commerceOrderItemId, commerceContext);
 			}
 			else if (cmd.equals(Constants.UPDATE)) {
 				int quantity = ParamUtil.getInteger(actionRequest, "quantity");

--- a/commerce-order-web/src/main/java/com/liferay/commerce/order/web/internal/portlet/action/EditCommerceOrderItemMVCActionCommand.java
+++ b/commerce-order-web/src/main/java/com/liferay/commerce/order/web/internal/portlet/action/EditCommerceOrderItemMVCActionCommand.java
@@ -81,6 +81,10 @@ public class EditCommerceOrderItemMVCActionCommand
 	protected void deleteCommerceOrderItems(ActionRequest actionRequest)
 		throws Exception {
 
+		CommerceContext commerceContext =
+			(CommerceContext)actionRequest.getAttribute(
+				CommerceWebKeys.COMMERCE_CONTEXT);
+
 		long[] deleteCommerceOrderItemIds = null;
 
 		long commerceOrderItemId = ParamUtil.getLong(
@@ -98,7 +102,7 @@ public class EditCommerceOrderItemMVCActionCommand
 
 		for (long deleteCommerceOrderItemId : deleteCommerceOrderItemIds) {
 			_commerceOrderItemService.deleteCommerceOrderItem(
-				deleteCommerceOrderItemId);
+				deleteCommerceOrderItemId, commerceContext);
 		}
 	}
 

--- a/commerce-order-web/src/main/resources/META-INF/resources/init.jsp
+++ b/commerce-order-web/src/main/resources/META-INF/resources/init.jsp
@@ -50,6 +50,7 @@ page import="com.liferay.commerce.order.web.security.permission.resource.Commerc
 page import="com.liferay.commerce.payment.model.CommercePaymentMethodGroupRel" %><%@
 page import="com.liferay.commerce.price.CommerceProductPrice" %><%@
 page import="com.liferay.commerce.product.model.CPInstance" %><%@
+page import="com.liferay.expando.kernel.model.ExpandoBridge" %><%@
 page import="com.liferay.petra.string.StringPool" %><%@
 page import="com.liferay.portal.kernel.bean.BeanParamUtil" %><%@
 page import="com.liferay.portal.kernel.dao.search.ResultRow" %><%@
@@ -67,10 +68,13 @@ page import="com.liferay.portal.kernel.util.Validator" %><%@
 page import="com.liferay.portal.kernel.util.WebKeys" %><%@
 page import="com.liferay.portal.kernel.workflow.WorkflowDefinition" %>
 
+<%@ page import="java.io.Serializable" %>
+
 <%@ page import="java.text.Format" %>
 
 <%@ page import="java.util.Date" %><%@
-page import="java.util.List" %>
+page import="java.util.List" %><%@
+page import="java.util.Map" %>
 
 <%@ page import="javax.portlet.PortletURL" %>
 

--- a/commerce-order-web/src/main/resources/META-INF/resources/order_item/detail.jsp
+++ b/commerce-order-web/src/main/resources/META-INF/resources/order_item/detail.jsp
@@ -35,6 +35,10 @@ portletDisplay.setURLBack(redirect);
 String title = LanguageUtil.format(request, "order-x", commerceOrder.getCommerceOrderId()) + " - " + HtmlUtil.escape(commerceOrderItem.getSku());
 
 renderResponse.setTitle(title);
+
+ExpandoBridge expandoBridge = commerceOrderItem.getExpandoBridge();
+
+Map<String, Serializable> attributes = expandoBridge.getAttributes();
 %>
 
 <portlet:actionURL name="editCommerceOrderItem" var="editCommerceOrderItemActionURL" />
@@ -57,12 +61,14 @@ renderResponse.setTitle(title);
 		</aui:fieldset>
 	</aui:fieldset-group>
 
-	<liferay-ui:form-navigator
-		formModelBean="<%= commerceOrderItem %>"
-		id="<%= CommerceOrderFormNavigatorConstants.FORM_NAVIGATOR_ID_COMMERCE_ORDER_ITEM_DETAILS %>"
-		markupView="lexicon"
-		showButtons="<%= false %>"
-	/>
+	<c:if test="<%= attributes.size() > 0 %>">
+		<liferay-ui:form-navigator
+			formModelBean="<%= commerceOrderItem %>"
+			id="<%= CommerceOrderFormNavigatorConstants.FORM_NAVIGATOR_ID_COMMERCE_ORDER_ITEM_DETAILS %>"
+			markupView="lexicon"
+			showButtons="<%= false %>"
+		/>
+	</c:if>
 
 	<aui:button-row>
 		<aui:button cssClass="btn-lg" type="submit" />

--- a/commerce-service/src/main/java/com/liferay/commerce/service/http/CommerceOrderItemServiceHttp.java
+++ b/commerce-service/src/main/java/com/liferay/commerce/service/http/CommerceOrderItemServiceHttp.java
@@ -120,6 +120,36 @@ public class CommerceOrderItemServiceHttp {
 		}
 	}
 
+	public static void deleteCommerceOrderItem(HttpPrincipal httpPrincipal,
+		long commerceOrderItemId,
+		com.liferay.commerce.context.CommerceContext commerceContext)
+		throws com.liferay.portal.kernel.exception.PortalException {
+		try {
+			MethodKey methodKey = new MethodKey(CommerceOrderItemServiceUtil.class,
+					"deleteCommerceOrderItem",
+					_deleteCommerceOrderItemParameterTypes2);
+
+			MethodHandler methodHandler = new MethodHandler(methodKey,
+					commerceOrderItemId, commerceContext);
+
+			try {
+				TunnelUtil.invoke(httpPrincipal, methodHandler);
+			}
+			catch (Exception e) {
+				if (e instanceof com.liferay.portal.kernel.exception.PortalException) {
+					throw (com.liferay.portal.kernel.exception.PortalException)e;
+				}
+
+				throw new com.liferay.portal.kernel.exception.SystemException(e);
+			}
+		}
+		catch (com.liferay.portal.kernel.exception.SystemException se) {
+			_log.error(se, se);
+
+			throw se;
+		}
+	}
+
 	public static com.liferay.commerce.model.CommerceOrderItem fetchByExternalReferenceCode(
 		HttpPrincipal httpPrincipal, long companyId,
 		String externalReferenceCode)
@@ -127,7 +157,7 @@ public class CommerceOrderItemServiceHttp {
 		try {
 			MethodKey methodKey = new MethodKey(CommerceOrderItemServiceUtil.class,
 					"fetchByExternalReferenceCode",
-					_fetchByExternalReferenceCodeParameterTypes2);
+					_fetchByExternalReferenceCodeParameterTypes3);
 
 			MethodHandler methodHandler = new MethodHandler(methodKey,
 					companyId, externalReferenceCode);
@@ -160,7 +190,7 @@ public class CommerceOrderItemServiceHttp {
 		try {
 			MethodKey methodKey = new MethodKey(CommerceOrderItemServiceUtil.class,
 					"fetchCommerceOrderItem",
-					_fetchCommerceOrderItemParameterTypes3);
+					_fetchCommerceOrderItemParameterTypes4);
 
 			MethodHandler methodHandler = new MethodHandler(methodKey,
 					commerceOrderItemId);
@@ -193,7 +223,7 @@ public class CommerceOrderItemServiceHttp {
 		try {
 			MethodKey methodKey = new MethodKey(CommerceOrderItemServiceUtil.class,
 					"getAvailableForShipmentCommerceOrderItems",
-					_getAvailableForShipmentCommerceOrderItemsParameterTypes4);
+					_getAvailableForShipmentCommerceOrderItemsParameterTypes5);
 
 			MethodHandler methodHandler = new MethodHandler(methodKey,
 					commerceOrderId);
@@ -225,7 +255,7 @@ public class CommerceOrderItemServiceHttp {
 		throws com.liferay.portal.kernel.exception.PortalException {
 		try {
 			MethodKey methodKey = new MethodKey(CommerceOrderItemServiceUtil.class,
-					"getCommerceOrderItem", _getCommerceOrderItemParameterTypes5);
+					"getCommerceOrderItem", _getCommerceOrderItemParameterTypes6);
 
 			MethodHandler methodHandler = new MethodHandler(methodKey,
 					commerceOrderItemId);
@@ -258,7 +288,7 @@ public class CommerceOrderItemServiceHttp {
 		try {
 			MethodKey methodKey = new MethodKey(CommerceOrderItemServiceUtil.class,
 					"getCommerceOrderItems",
-					_getCommerceOrderItemsParameterTypes6);
+					_getCommerceOrderItemsParameterTypes7);
 
 			MethodHandler methodHandler = new MethodHandler(methodKey,
 					commerceOrderId, start, end);
@@ -291,7 +321,7 @@ public class CommerceOrderItemServiceHttp {
 		try {
 			MethodKey methodKey = new MethodKey(CommerceOrderItemServiceUtil.class,
 					"getCommerceOrderItemsCount",
-					_getCommerceOrderItemsCountParameterTypes7);
+					_getCommerceOrderItemsCountParameterTypes8);
 
 			MethodHandler methodHandler = new MethodHandler(methodKey,
 					commerceOrderId);
@@ -324,7 +354,7 @@ public class CommerceOrderItemServiceHttp {
 		try {
 			MethodKey methodKey = new MethodKey(CommerceOrderItemServiceUtil.class,
 					"getCommerceOrderItemsQuantity",
-					_getCommerceOrderItemsQuantityParameterTypes8);
+					_getCommerceOrderItemsQuantityParameterTypes9);
 
 			MethodHandler methodHandler = new MethodHandler(methodKey,
 					commerceOrderId);
@@ -358,7 +388,7 @@ public class CommerceOrderItemServiceHttp {
 		try {
 			MethodKey methodKey = new MethodKey(CommerceOrderItemServiceUtil.class,
 					"getCommerceWarehouseItemQuantity",
-					_getCommerceWarehouseItemQuantityParameterTypes9);
+					_getCommerceWarehouseItemQuantityParameterTypes10);
 
 			MethodHandler methodHandler = new MethodHandler(methodKey,
 					commerceOrderItemId, commerceWarehouseId);
@@ -391,7 +421,7 @@ public class CommerceOrderItemServiceHttp {
 		throws com.liferay.portal.kernel.exception.PortalException {
 		try {
 			MethodKey methodKey = new MethodKey(CommerceOrderItemServiceUtil.class,
-					"search", _searchParameterTypes10);
+					"search", _searchParameterTypes11);
 
 			MethodHandler methodHandler = new MethodHandler(methodKey,
 					commerceOrderId, keywords, start, end, sort);
@@ -425,7 +455,7 @@ public class CommerceOrderItemServiceHttp {
 		throws com.liferay.portal.kernel.exception.PortalException {
 		try {
 			MethodKey methodKey = new MethodKey(CommerceOrderItemServiceUtil.class,
-					"search", _searchParameterTypes11);
+					"search", _searchParameterTypes12);
 
 			MethodHandler methodHandler = new MethodHandler(methodKey,
 					commerceOrderId, sku, name, andOperator, start, end, sort);
@@ -460,7 +490,7 @@ public class CommerceOrderItemServiceHttp {
 		try {
 			MethodKey methodKey = new MethodKey(CommerceOrderItemServiceUtil.class,
 					"updateCommerceOrderItem",
-					_updateCommerceOrderItemParameterTypes12);
+					_updateCommerceOrderItemParameterTypes13);
 
 			MethodHandler methodHandler = new MethodHandler(methodKey,
 					commerceOrderItemId, quantity, commerceContext,
@@ -497,7 +527,7 @@ public class CommerceOrderItemServiceHttp {
 		try {
 			MethodKey methodKey = new MethodKey(CommerceOrderItemServiceUtil.class,
 					"updateCommerceOrderItem",
-					_updateCommerceOrderItemParameterTypes13);
+					_updateCommerceOrderItemParameterTypes14);
 
 			MethodHandler methodHandler = new MethodHandler(methodKey,
 					commerceOrderItemId, quantity, json, commerceContext,
@@ -534,7 +564,7 @@ public class CommerceOrderItemServiceHttp {
 		try {
 			MethodKey methodKey = new MethodKey(CommerceOrderItemServiceUtil.class,
 					"upsertCommerceOrderItem",
-					_upsertCommerceOrderItemParameterTypes14);
+					_upsertCommerceOrderItemParameterTypes15);
 
 			MethodHandler methodHandler = new MethodHandler(methodKey,
 					commerceOrderId, cpInstanceId, quantity, shippedQuantity,
@@ -571,45 +601,48 @@ public class CommerceOrderItemServiceHttp {
 	private static final Class<?>[] _deleteCommerceOrderItemParameterTypes1 = new Class[] {
 			long.class
 		};
-	private static final Class<?>[] _fetchByExternalReferenceCodeParameterTypes2 =
+	private static final Class<?>[] _deleteCommerceOrderItemParameterTypes2 = new Class[] {
+			long.class, com.liferay.commerce.context.CommerceContext.class
+		};
+	private static final Class<?>[] _fetchByExternalReferenceCodeParameterTypes3 =
 		new Class[] { long.class, String.class };
-	private static final Class<?>[] _fetchCommerceOrderItemParameterTypes3 = new Class[] {
+	private static final Class<?>[] _fetchCommerceOrderItemParameterTypes4 = new Class[] {
 			long.class
 		};
-	private static final Class<?>[] _getAvailableForShipmentCommerceOrderItemsParameterTypes4 =
+	private static final Class<?>[] _getAvailableForShipmentCommerceOrderItemsParameterTypes5 =
 		new Class[] { long.class };
-	private static final Class<?>[] _getCommerceOrderItemParameterTypes5 = new Class[] {
+	private static final Class<?>[] _getCommerceOrderItemParameterTypes6 = new Class[] {
 			long.class
 		};
-	private static final Class<?>[] _getCommerceOrderItemsParameterTypes6 = new Class[] {
+	private static final Class<?>[] _getCommerceOrderItemsParameterTypes7 = new Class[] {
 			long.class, int.class, int.class
 		};
-	private static final Class<?>[] _getCommerceOrderItemsCountParameterTypes7 = new Class[] {
+	private static final Class<?>[] _getCommerceOrderItemsCountParameterTypes8 = new Class[] {
 			long.class
 		};
-	private static final Class<?>[] _getCommerceOrderItemsQuantityParameterTypes8 =
+	private static final Class<?>[] _getCommerceOrderItemsQuantityParameterTypes9 =
 		new Class[] { long.class };
-	private static final Class<?>[] _getCommerceWarehouseItemQuantityParameterTypes9 =
+	private static final Class<?>[] _getCommerceWarehouseItemQuantityParameterTypes10 =
 		new Class[] { long.class, long.class };
-	private static final Class<?>[] _searchParameterTypes10 = new Class[] {
+	private static final Class<?>[] _searchParameterTypes11 = new Class[] {
 			long.class, String.class, int.class, int.class,
 			com.liferay.portal.kernel.search.Sort.class
 		};
-	private static final Class<?>[] _searchParameterTypes11 = new Class[] {
+	private static final Class<?>[] _searchParameterTypes12 = new Class[] {
 			long.class, String.class, String.class, boolean.class, int.class,
 			int.class, com.liferay.portal.kernel.search.Sort.class
 		};
-	private static final Class<?>[] _updateCommerceOrderItemParameterTypes12 = new Class[] {
+	private static final Class<?>[] _updateCommerceOrderItemParameterTypes13 = new Class[] {
 			long.class, int.class,
 			com.liferay.commerce.context.CommerceContext.class,
 			com.liferay.portal.kernel.service.ServiceContext.class
 		};
-	private static final Class<?>[] _updateCommerceOrderItemParameterTypes13 = new Class[] {
+	private static final Class<?>[] _updateCommerceOrderItemParameterTypes14 = new Class[] {
 			long.class, int.class, String.class,
 			com.liferay.commerce.context.CommerceContext.class,
 			com.liferay.portal.kernel.service.ServiceContext.class
 		};
-	private static final Class<?>[] _upsertCommerceOrderItemParameterTypes14 = new Class[] {
+	private static final Class<?>[] _upsertCommerceOrderItemParameterTypes15 = new Class[] {
 			long.class, long.class, int.class, int.class, String.class,
 			com.liferay.commerce.context.CommerceContext.class,
 			com.liferay.portal.kernel.service.ServiceContext.class

--- a/commerce-service/src/main/java/com/liferay/commerce/service/http/CommerceOrderItemServiceSoap.java
+++ b/commerce-service/src/main/java/com/liferay/commerce/service/http/CommerceOrderItemServiceSoap.java
@@ -97,6 +97,20 @@ public class CommerceOrderItemServiceSoap {
 		}
 	}
 
+	public static void deleteCommerceOrderItem(long commerceOrderItemId,
+		com.liferay.commerce.context.CommerceContext commerceContext)
+		throws RemoteException {
+		try {
+			CommerceOrderItemServiceUtil.deleteCommerceOrderItem(commerceOrderItemId,
+				commerceContext);
+		}
+		catch (Exception e) {
+			_log.error(e, e);
+
+			throw new RemoteException(e.getMessage());
+		}
+	}
+
 	public static com.liferay.commerce.model.CommerceOrderItemSoap fetchByExternalReferenceCode(
 		long companyId, String externalReferenceCode) throws RemoteException {
 		try {

--- a/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderItemLocalServiceImpl.java
+++ b/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderItemLocalServiceImpl.java
@@ -143,9 +143,17 @@ public class CommerceOrderItemLocalServiceImpl
 
 		commerceOrderItemPersistence.update(commerceOrderItem);
 
+		commerceOrderLocalService.recalculatePrice(
+			commerceOrderItem.getCommerceOrderId(), commerceContext);
+
 		return commerceOrderItem;
 	}
 
+	/**
+	 * @return
+	 * @deprecated As of Mueller (7.2.x), use deleteCommerceOrderItem(CommerceOrderItem, CommerceContext)
+	 */
+	@Deprecated
 	@Indexable(type = IndexableType.DELETE)
 	@Override
 	public CommerceOrderItem deleteCommerceOrderItem(
@@ -160,6 +168,28 @@ public class CommerceOrderItemLocalServiceImpl
 
 		expandoRowLocalService.deleteRows(
 			commerceOrderItem.getCommerceOrderItemId());
+
+		return commerceOrderItem;
+	}
+
+	@Indexable(type = IndexableType.DELETE)
+	@Override
+	public CommerceOrderItem deleteCommerceOrderItem(
+			CommerceOrderItem commerceOrderItem,
+			CommerceContext commerceContext)
+		throws PortalException {
+
+		// Commerce order item
+
+		commerceOrderItemPersistence.remove(commerceOrderItem);
+
+		// Expando
+
+		expandoRowLocalService.deleteRows(
+			commerceOrderItem.getCommerceOrderItemId());
+
+		commerceOrderLocalService.recalculatePrice(
+			commerceOrderItem.getCommerceOrderId(), commerceContext);
 
 		return commerceOrderItem;
 	}
@@ -387,6 +417,9 @@ public class CommerceOrderItemLocalServiceImpl
 			commerceOrderItem, commerceProductPrice.getDiscountValue());
 
 		commerceOrderItemPersistence.update(commerceOrderItem);
+
+		commerceOrderLocalService.recalculatePrice(
+			commerceOrderItem.getCommerceOrderId(), commerceContext);
 
 		return commerceOrderItem;
 	}

--- a/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderItemServiceImpl.java
+++ b/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderItemServiceImpl.java
@@ -66,6 +66,23 @@ public class CommerceOrderItemServiceImpl
 	}
 
 	@Override
+	public void deleteCommerceOrderItem(
+			long commerceOrderItemId, CommerceContext commerceContext)
+		throws PortalException {
+
+		CommerceOrderItem commerceOrderItem =
+			commerceOrderItemLocalService.getCommerceOrderItem(
+				commerceOrderItemId);
+
+		_commerceOrderModelResourcePermission.check(
+			getPermissionChecker(), commerceOrderItem.getCommerceOrderId(),
+			ActionKeys.UPDATE);
+
+		commerceOrderItemLocalService.deleteCommerceOrderItem(
+			commerceOrderItem, commerceContext);
+	}
+
+	@Override
 	public CommerceOrderItem fetchByExternalReferenceCode(
 			long companyId, String externalReferenceCode)
 		throws PortalException {

